HID-1813: suppress modal when list isn't found

rupl opened the pull request:
## https://humanitarian.atlassian.net/browse/HID-1813

The UX of the "list not found" page is better than the modal itself, so we suppress showing it and rely only on the template to direct a visitor to the next step.

@guillaumev I believe I got the conditions for this API request pretty specific but would love your 👀 on it to ensure I didn't accidentally suppress modals on other situations.

rupl committed:
HID-1813: suppress modal when list isn't found

The template shows an informative message with links to
continue searching, so the modal just got in the way.

guillaumev commented on src/app/common/errors.controller.js:
```diff
@@ -46,6 +50,12 @@
 
+      // List not found — the template shows a good error message so we will
+      // suppress the modal from displaying.
+      if (rejection.config.method === 'GET' && rejection.config.url.indexOf('/list/')) {
```
@rupl I'm thinking maybe it would be better to use HTTP status codes here, and display a modal only when status !== 404 && status !== 400 ?

rupl replied:
Sure I can explicitly check codes. However, the ticket was created specifically to deal with a 500 coming back. That's why I want to be careful and not under-report 500s that maybe do need to be shown to the user (in cases where a template isn't providing context).

rupl replied:
Come to think of it, we do want to suppress the modal when it's 404. The page looks like this already — spacing fixed in #175

![image](https://user-images.githubusercontent.com/254753/66482651-b3e1cf80-eaa3-11e9-9175-11178a3ca05a.png)

rupl committed:
HID-1813: adjust conditional based on HID API changes

See UN-OCHA/hid_api#44

rupl commented:
@guillaumev ready for review again

rupl committed:
HID-1813: more thorough conditional based on HID API changes

The API now outputs both 400 and 404 based on whether the ID string
is valid or not.
